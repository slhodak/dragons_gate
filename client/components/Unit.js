import React from 'react';
import '../style.css';

// Unit
export default (props) => {
  const {
    healthPoints,
    speed,
    meleeRange,
    meleeDamage,
    rangedRange,
    rangedDamage,
    defenseArmor,
    healthRegen,
    status,
    name,
    id
  } = props.unit;

  return (
    <div className="unit" onClick={() => { props.selectCombatant({ name, id }) }}>
      <h3>{name}</h3>
      <div>{`ID: ${id}`}</div>
      <div><strong>{status}</strong></div>
      <div>{`HP: ${healthPoints}`}</div>
      <div>{`Speed: ${speed}`}</div>
      <div>{`Melee Range: ${meleeRange}`}</div>
      <div>{`Melee Damage: ${meleeDamage}`}</div>
      <div>{`Ranged Range: ${rangedRange}`}</div>
      <div>{`Ranged Damage: ${rangedDamage}`}</div>
      <div>{`Defense Armor: ${defenseArmor}`}</div>
      <div>{`Health Regen: ${healthRegen}`}</div>
    </div>
  )
}
