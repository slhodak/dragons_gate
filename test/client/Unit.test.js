import React from 'react';
import renderer from 'react-test-renderer';
import Unit from '../../client/components/Unit.js';
import { mockEmpire } from './__mocks__/factionMocks.js';

test('it should render correctly', () => {
  const eliteSoldier = mockEmpire.units[0];
  const tree = renderer
    .create(<Unit unit={eliteSoldier} myTurn={false} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
