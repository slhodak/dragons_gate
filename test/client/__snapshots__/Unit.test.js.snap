// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should render correctly 1`] = `
<div
  className="unit"
>
  <div
    className="header"
  >
    <h3>
      Elite Soldier
    </h3>
    <span
      className="healthy"
    >
      <em>
        healthy
      </em>
    </span>
  </div>
  <div>
    HP: 30
    <span
      className="loss false"
    />
  </div>
  <div>
    Speed: 3
  </div>
  <div>
    Melee Range: 2
  </div>
  <div>
    Melee Damage: 4,4
    <div
      className="combatButton"
    />
  </div>
  <div>
    Defense Armor: 2,4
  </div>
  <div>
    Health Regen: 2
  </div>
  <div
    className="combatButton"
  />
</div>
`;
